src/components/ui/Button/IconButton.tsx

```diff
@@ -3,12 +3,19 @@
 
 import React, { forwardRef } from 'react';
 import Button, { ButtonProps } from './';
+import { StyledIconButtonWrapper } from './Styled';
 
-export interface IconButtonProps extends ButtonProps {}
+export interface IconButtonProps extends ButtonProps {
+  /** Render a component to the top right area of the IconButton */
+  badge?: React.ReactNode | React.ReactNode[];
+}
 
 export const IconButton = forwardRef(
   (props: IconButtonProps, ref: React.Ref<HTMLButtonElement>) => (
-    <Button ref={ref} variant="icon" {...props} />
+    <StyledIconButtonWrapper>
+      <Button ref={ref} variant="icon" {...props} />
+      {props.badge && props.badge}
+    </StyledIconButtonWrapper>
```

Comment:
Do you foresee any isues with spreading props to the inner component and not the wrapper? Might get some weird behavior with something like `<IconButton className="pos-absolute" />`, since it'll be applied to the button but not the wrapper

Since it's pretty easy to work around, not a big dea but figured I'd bring it up